zaza/openstack/charm_tests/ceph/tests.py

```diff
@@ -545,6 +546,21 @@ def test_list_disks(self):
         )
         logging.debug('OK')
 
+    def test_cache_device(self):
+        """Test adding a new disk with a caching device."""
+        logging.info('Running add-disk action with a caching device')
+        osds = [x.entity_id for x in zaza_model.get_units('ceph-osd')]
+        for unit in osds:
+            zaza_juju.add_storage(unit, 'cache-devices', 'cinder', 10)
+            loop_dev = zaza_juju.add_loop_device(unit, 10)
+            action_obj = zaza_model.run_action(
+                unit_name=unit,
+                action_name='add-disk',
+                action_params={'osd-devices': loop_dev.get('Stdout'),
+                               'partition-size': 5}
+            )
+            zaza_utils.assertActionRanOK(action_obj)
+
```

Comment:
It would be good to also verify that a new OSD has been added to the Ceph cluster.